modules/__tests__/Link-modular-test.js

```diff
@@ -0,0 +1,284 @@
+    var App = React.createClass({
+      render() {
+        return (
+          <div>
+            <Link to="/hello/michael">Michael</Link>
+            <Link to="/hello/ryan">Ryan</Link>
+          </div>
+        );
+      }
+    });
+
+    it('is active when its params match', function (done) {
+      React.render((
+        <Router history={createHistory('/hello/michael')}>
+          <Route path="/" component={App}>
+            <Route path="hello/:name" component={Hello}/>
+          </Route>
+        </Router>
+      ), node, function () {
+        var a = node.querySelectorAll('a')[0];
+        expect(a.className.trim()).toEqual('active');
+        done();
+      });
+    });
+
+    it('is not active when its params do not match', function (done) {
+      React.render((
+        <Router history={createHistory('/hello/michael')}>
+          <Route path="/" component={App}>
+            <Route path="hello/:name" component={Hello}/>
+          </Route>
+        </Router>
+      ), node, function () {
+        var a = node.querySelectorAll('a')[1];
+        expect(a.className.trim()).toEqual('');
+        done();
+      });
+    });
+  });
+
+  describe('when its route is active and className is empty', function () {
+    it('it shouldn\'t have an active class', function (done) {
+      var LinkWrapper = React.createClass({
+        render() {
+          return (
+            <div>
+              <Link to="/hello" className="dontKillMe" activeClassName="">Link</Link>
+              {this.props.children}
+            </div>
+          );
+        }
+      });
+
+      var a, steps = [
+        function () {
+          a = node.querySelector('a');
+          expect(a.className).toEqual('dontKillMe');
+          this.transitionTo('/hello');
+        },
+        function () {
+          expect(a.className).toEqual('dontKillMe');
+        }
+      ];
+
+      var execNextStep = execSteps(steps, done);
+
+      React.render((
+        <Router history={createHistory('/goodbye')} onUpdate={execNextStep}>
+          <Route path="/" component={LinkWrapper}>
+            <Route path="goodbye" component={Goodbye}/>
+            <Route path="hello" component={Hello}/>
+          </Route>
+        </Router>
+      ), node, execNextStep);
+    });
+  });
+
+  describe('when its route is active', function () {
+    it('has its activeClassName', function (done) {
+      var LinkWrapper = React.createClass({
+        render() {
+          return (
+            <div>
+              <Link to="/hello" className="dontKillMe" activeClassName="highlight">Link</Link>
+              {this.props.children}
+            </div>
+          );
+        }
+      });
+
+      var a, steps = [
+        function () {
+          a = node.querySelector('a');
+          expect(a.className).toEqual('dontKillMe');
+          this.transitionTo('/hello');
+        },
+        function () {
+          a = node.querySelector('a'); // TODO: Why is this necessary?
```

Comment:
Ahh my hunch is that it's because the tree was re-built instead of updated. I think it may have to do with the dynamically-created component inside Router-modular (when we apply the HoC). Should be an easy fix, I'll try tomorrow.